[Flaky test] TestController_RotateCertificates

I have seen the test fail in CI (`Unit test (ubuntu-latest)`), with the following logs:

```
2022-09-13T22:07:23.5780808Z W0913 22:07:12.163091   23569 env.go:88] Environment variable POD_NAMESPACE not found
2022-09-13T22:07:23.5782372Z W0913 22:07:12.163183   23569 env.go:126] Failed to get Pod Namespace from environment. Using "kube-system" as the Antrea Service Namespace
2022-09-13T22:07:23.5783304Z E0913 22:07:12.560524   23569 ipsec_certificate_controller.go:261] "Verifying current certificate configurations failed" err="certificate and key pair is nil"
2022-09-13T22:07:23.5784002Z I0913 22:07:12.560579   23569 ipsec_certificate_controller.go:268] "Start rotating IPsec certificate"
2022-09-13T22:07:23.5784733Z E0913 22:07:12.744620   23569 ipsec_certificate_controller.go:261] "Verifying current certificate configurations failed" err="CA certificate is empty"
2022-09-13T22:07:23.5785408Z I0913 22:07:12.744669   23569 ipsec_certificate_controller.go:268] "Start rotating IPsec certificate"
2022-09-13T22:07:23.5786114Z E0913 22:07:13.428822   23569 ipsec_certificate_controller.go:261] "Verifying current certificate configurations failed" err="certificate and key pair is nil"
2022-09-13T22:07:23.5787053Z I0913 22:07:13.428877   23569 ipsec_certificate_controller.go:268] "Start rotating IPsec certificate"
2022-09-13T22:07:23.5787759Z E0913 22:07:13.749592   23569 ipsec_certificate_controller.go:261] "Verifying current certificate configurations failed" err="certificate and key pair is nil"
2022-09-13T22:07:23.5788477Z I0913 22:07:13.749644   23569 ipsec_certificate_controller.go:268] "Start rotating IPsec certificate"
2022-09-13T22:07:23.5789721Z I0913 22:07:14.922615   23569 ipsec_certificate_controller.go:471] "Created new certificate and key for IPSec" cert="/tmp/antrea-ipsec-test169962986/fake-node-1-9ebb1176b6.crt" key="/tmp/antrea-ipsec-test169962986/fake-node-1-9ebb1176b6.key"
2022-09-13T22:07:23.5791031Z I0913 22:07:14.927151   23569 ipsec_certificate_controller.go:223] "Calculated certificate rotation deadline" expiration="2022-09-14 22:07:13 +0000 UTC" deadline="2022-09-14 16:51:58.879519679 +0000 UTC"
2022-09-13T22:07:23.5791855Z I0913 22:07:14.927249   23569 ipsec_certificate_controller.go:383] "Updating OVS configurations for IPsec certificates"
2022-09-13T22:07:23.5792491Z I0913 22:07:14.927841   23569 ipsec_certificate_controller.go:383] "Updating OVS configurations for IPsec certificates"
2022-09-13T22:07:23.5793143Z I0913 22:07:15.240443   23569 ipsec_certificate_controller.go:411] "Starting AntreaAgentIPsecCertificateController"
2022-09-13T22:07:23.5793850Z E0913 22:07:15.240612   23569 ipsec_certificate_controller.go:261] "Verifying current certificate configurations failed" err="certificate and key pair is nil"
2022-09-13T22:07:23.5794487Z I0913 22:07:15.240671   23569 ipsec_certificate_controller.go:268] "Start rotating IPsec certificate"
2022-09-13T22:07:23.5795589Z I0913 22:07:16.608027   23569 ipsec_certificate_controller.go:471] "Created new certificate and key for IPSec" cert="/tmp/antrea-ipsec-test1602412469/fake-node-1-20f4979105.crt" key="/tmp/antrea-ipsec-test1602412469/fake-node-1-20f4979105.key"
2022-09-13T22:07:23.5796733Z I0913 22:07:16.609320   23569 ipsec_certificate_controller.go:223] "Calculated certificate rotation deadline" expiration="2022-09-13 22:07:25 +0000 UTC" deadline="2022-09-13 22:07:22.030147099 +0000 UTC"
2022-09-13T22:07:23.5797415Z I0913 22:07:16.609408   23569 ipsec_certificate_controller.go:383] "Updating OVS configurations for IPsec certificates"
2022-09-13T22:07:23.5797999Z I0913 22:07:22.031616   23569 ipsec_certificate_controller.go:268] "Start rotating IPsec certificate"
2022-09-13T22:07:23.5798584Z --- FAIL: TestController_RotateCertificates (8.33s)
2022-09-13T22:07:23.5799367Z     ipsec_certificate_controller_test.go:382: Sign CSR "fake-node-1-qzfspc5x" successfully
2022-09-13T22:07:23.5800191Z     ipsec_certificate_controller_test.go:382: Sign CSR "fake-node-1-f8pbjlnq" successfully
2022-09-13T22:07:23.5800761Z     ipsec_certificate_controller_test.go:317: 
2022-09-13T22:07:23.5801690Z         	Error Trace:	/home/runner/work/antrea/antrea/pkg/agent/controller/ipseccertificate/ipsec_certificate_controller_test.go:317
2022-09-13T22:07:23.5802446Z         	Error:      	"7s" is not less than or equal to "6.656622822s"
2022-09-13T22:07:23.5803757Z         	Test:       	TestController_RotateCertificates
2022-09-13T22:07:23.5804772Z I0913 22:07:23.261391   23569 ipsec_certificate_controller.go:471] "Created new certificate and key for IPSec" cert="/tmp/antrea-ipsec-test1602412469/fake-node-1-d174243e57.crt" key="/tmp/antrea-ipsec-test1602412469/fake-node-1-d174243e57.key"
2022-09-13T22:07:23.5805834Z I0913 22:07:23.262802   23569 ipsec_certificate_controller.go:223] "Calculated certificate rotation deadline" expiration="2022-09-13 22:07:32 +0000 UTC" deadline="2022-09-13 22:07:30.793340975 +0000 UTC"
2022-09-13T22:07:23.5806446Z I0913 22:07:23.262865   23569 ipsec_certificate_controller.go:383] "Updating OVS configurations for IPsec certificates"
2022-09-13T22:07:23.5807015Z I0913 22:07:23.262923   23569 ipsec_certificate_controller.go:435] "Shutting down AntreaAgentIPsecCertificateController"
2022-09-13T22:07:23.5808078Z I0913 22:07:23.263657   23569 ipsec_certificate_controller.go:223] "Calculated certificate rotation deadline" expiration="2022-05-20 10:00:00 +0000 UTC" deadline="2022-05-20 08:03:18.342424885 +0000 UTC"
2022-09-13T22:07:23.5809006Z I0913 22:07:23.263795   23569 ipsec_certificate_controller.go:223] "Calculated certificate rotation deadline" expiration="2022-05-30 00:00:00 +0000 UTC" deadline="2022-05-28 03:14:17.302095332 +0000 UTC"
2022-09-13T22:07:23.5809401Z FAIL
```

--- a/pkg/agent/controller/ipseccertificate/ipsec_certificate_controller_test.go
+++ b/pkg/agent/controller/ipseccertificate/ipsec_certificate_controller_test.go
@@ -311,10 +311,16 @@ func TestController_RotateCertificates(t *testing.T) {
 	if delta < 0 {
 		delta = -delta
 	}
-	// the rotation interval should be in [7s, 9s], but it takes time to process the CSR request,
-	// so add one second to the upper bound.
-	assert.Less(t, delta, time.Second*10)
-	assert.LessOrEqual(t, time.Second*7, delta)
+	// the rotation interval is determined by nextRotationDeadline as notBefore + (notAfter -
+	// notBefore) * k, where k is >= 0.7 and <= 0.9. We would therefore expect the delta to
+	// fall in the interval [7s, 9s]. However we have to account for the following:
+	// a) the accuracy of notAfter is at the second level, which means that it will be
+	// truncated, which means that there can actually be less than 7s between the
+	// CreationTimestamp of the first certificate and the rotation deadline. As a result, we
+	// need to set the lower bound to 6s.
+	// b) it takes time to process the CSR request so we add one second to the upper bound.
+	assert.Less(t, delta, 10*time.Second)
+	assert.LessOrEqual(t, 6*time.Second, delta)
 }
 
 func newIPsecCertTemplate(t *testing.T, nodeName string, notBefore, notAfter time.Time) *x509.Certificate {

